relation_engine_server/utils/arango_client.py

```diff
@@ -170,6 +268,74 @@ def import_from_file(file_path, query):
     return resp_json
 
 
+def get_all_views():
+    """
+    Fetch all existing views from server
+
+    Resp to GET /_api/view is
+    {
+        "error": false,
+        "code": 200,
+        "result": [
+            {"id": str, "name": str, "type": str},
+            ...
+        ]
+    }
+
+    Resp to GET /_api/view/{view_name}/properties is
+    {
+        "error" : false,
+        "code" : 200,
+        "writebufferIdle" : 64,
+        "type" : "arangosearch",
+        "writebufferSizeMax" : 33554432,
+        "consolidationPolicy" : {
+            "type" : "tier",
+            "segmentsBytesFloor" : 2097152,
+            "segmentsBytesMax" : 5368709120,
+            "segmentsMax" : 10,
+            "segmentsMin" : 1,
+            "minScore" : 0
+        },
+        "name" : "products",
+        "primarySort" : [ ],
+        "globallyUniqueId" : "hA5F3C05BE80C/68910",
+        "id" : "68910",
+        "storedValues" : [ ],
+        "writebufferActive" : 0,
+        "consolidationIntervalMsec" : 1000,
+        "cleanupIntervalStep" : 2,
+        "commitIntervalMsec" : 1000,
+        "links" : {
+        },
+        "primarySortCompression" : "lz4"
+    }
+
+    Returns
+    [
+        {},
+        {},
+        ...
+    ]
+    where each item is the properties dict (from above)
+    """
+    resp_json = adb_request(
+        req_method=requests.get,
+        url_append="/view",
+    )
+    view_names = [view["name"] for view in resp_json["result"]]
+
+    view_properties = []
+    for view_name in view_names:
```

Comment:
since you're not using `view_names`, you can iterate directly through `resp_json["result"]` to populate `view_properties`

Reply:
I think this was more for prose reasons ... the whole thing was a mess of differently structured JSON objects and sometimes it's nice to write things out or pretend you're following a pattern with the other functions (which I don't think I really am here ...)

Reply:
Yes, I was aware of what you were getting at. I don't think it's worth gunning for. It's 1 extra variable name, and I think it may vary by programmer which is actually more readable. IMO sometimes writing more complicated code to get minimally compact code does not provide meaningful performance gains or increase readability

Reply:
If you want to improve readability, move the lines involving `view_names` together.

The list comprehension is pointless as it adds to the visual and cognitive complexity without improving computational performance (which would be a micro-optimisation anyway). 🤷